server/text_generation_server/layers/gptq/__init__.py

```diff
@@ -321,7 +325,7 @@ def get_weights_row(self, weights: Weights, prefix: str):
             if g_idx is not None:
                 if (
                     not torch.equal(
-                        g_idx.cpu(),
+                        (g_idx - g_idx[0]).cpu(),
```

Comment:
This block needs a comment what is computed here. The old code was already difficult to read, but I the way I read it is that it is checking whether `g_idx` incrementing indices by group (so there is a `g_idx`, but no activation sorting).

But what is this subtraction doing?

Reply:
I think it's subtracting the first index, because say we have `g_idx` with `groupsize = 2`

```
[0 0 1 1 2 2 3 3]
```

If we have two shards, then it gets broken up into

```
[0 0 1 1] [2 2 3 3]
```

And then the check fails for the second shard. Subtracting `g_idx[0]` then changes this into

```
[0 0 1 1] [0 0 1 1]
```

Making the check valid. But there should be a comment to make it easier to understand without knowing all the context.

Reply:
YEs this is the core idea of the fix. The logic was only working on TP=1, it's now working on TP>1 (to detect that g_idx is redundant and can be safely ignored)

Reply:
Yeah makes sense, would be a good to have a short comment here though so that future-us don't have to re-parse what is going on here.